### PR TITLE
Adds CavcCorrespondenceMailTask 

### LINKED ISSUE #15239: New task type (Mail Team): 90 Day Letter Correspondence
## User or job story
User story: As a Mail Team user, I need to be able to create a new task on a CAVC Remand case, so that when I receive correspondence from an appellant I can send the correspondence to the CAVC Litigation Support Team. 

## Acceptance criteria
- [x] Please put this work behind the feature toggle: cavc_remand
- [x] This feature should be accessible to the following user groups: Mail Team
- [x] Other conditions:
  - [x] `CavcTask` is open
  - [x] Appeal is a cavc remand
- [ ] Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Update documentation: CAVC Remand
- [x] ~Task is child of CAVC task~
- [x] Verify "Add New Task" link is displayed on the appeal case details page
- [x] Verify Select Correspondence Type drop down includes CAVC Correspondence
- [x] Verify the new task 'CavcCorrespondenceTask' is routed to the CAVC Litigation Support team
- [x] Verify `CavcRemandProcessedLetterResponseWindowTask` status does not change. #15288

## Open questions
- [ ] With the assumption that the correspondence is NOT related to the response window and is just standard mail, what task actions should the `CavcCorrespondenceTask` have?

### Technical notes

This could possibly be similar to the Evidence Submission Task as they are both timed responses.

### PR DESCRIPTION
**Pull request by lomky**
Resolves #15239 

### Description
Creates the CavcCorrespondenceMailTask & its behavior.

### Acceptance Criteria
- [ ] work behind the feature toggle: cavc_remand
- [ ] Add task should be accessible to the Mail Team
- [ ] accessible to the following user groups: Mail Team
- [ ] Select Correspondence Type drop down includes CAVC Correspondence
- [ ] requires a CAVC Remand appeal with an open CavcTask
- [ ] Mail Team users have no actions on the created tasks
- [ ] Cavc Lit Admins can act on Organization task or any User task
- [ ] Cavc Lit user can act on a user task assigned to themselves
- [ ]  Verify CavcRemandProcessedLetterResponseWindowTask status does not change.

### Testing Plan
  1. Reset the database with `make reset`
  2. Create a Cavc Appeal at CavcResponseWindow
```ruby
appeal = FactoryBot.create(:appeal, :cavc_response_window_open)
appeal.reload.treee
appeal.uuid
FeatureToggle.disable!(:cavc_remand)
```
  3. Load up [caseflow](http://localhost:3000/test/users) and sign in as Jolly Postman
  4. Navigate to the appeal you created
  5. Click `+ Add a new task`
   - [x] CAVC Correspondence is not available
  6. Enable the flag: `FeatureToggle.enable!(:cavc_remand)`
  7. Reload and Click `+ Add a new task`
   - [x] CAVC Correspondence is available
   - [x] Creating a CAVC Correspondence task autoassigns to CAVC Lit Support
   - [x] No actions are available on the task for the mail user
  8. Change to be a CAVC Lit Support admin user
  9. Navigate to the appeal
   - [x] Check the actions exist and do what you want (TK)
  10. Create another appeal & mail task
  11. Navigate to the appeal and assign it to a Cavc Lit support user
   - [x] confirm you have admin have actions on the user task
  12. Sign in as a Cavc Lit user you didn't assign the task to 
   - [x] No actions are available
  13. Sign in as the assigned Cavc Lit user
   - [x] Check the actions exist and do what you want (TK)

Expected Behavior on Actions:
  - [x] Confirm none of this affects the CavcResponseWindowTask status!
  - [x] Change Task Type: Cancels the CavcCorrespondence subtree, creates a sibling of the new type
  - [x] (Re)Assign to person: creates a user task
  - [x] Place on hold: places a hold
  - [x] Mark task complete: completes the subtree
  - [x] Cancel task: cancels the subtree
 
### Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### UI Changes
image

### CODE REVIEW

File: app/models/tasks/mail_task.rb [hunk 1]
```diff
@@ -72,11 +72,6 @@ def child_task_assignee(parent, params)
       end
     end
 
-    def outstanding_cavc_tasks?(_parent)
-      # We don't yet create CAVC tasks so this will always return false for now.
-      false
-    end
-
```
Comment (lomky) [inline app/models/tasks/mail_task.rb:79]: Unused function written 2 years ago that no longer jives with our implementation of cavc appeal streams.

File: spec/models/tasks/mail_task_spec.rb [hunk 2]
```diff
@@ -101,12 +101,6 @@
     end
   end
 
-  describe ".outstanding_cavc_tasks?" do
-    it "should always return false since we do not yet have CAVC tasks" do
-      expect(MailTask.outstanding_cavc_tasks?(root_task)).to eq(false)
-    end
-  end
-
```
Comment (lomky) [inline spec/models/tasks/mail_task_spec.rb:109]: unused method's spec :yeet:

File: app/models/tasks/cavc_correspondence_mail_task.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+##
+# Task created by the mail team when mail arrives for a CAVC Appeal is in processing with the CAVC Litigation Support
```
Comment (n1zyy) [inline app/models/tasks/cavc_correspondence_mail_task.rb:4]:
```suggestion
# Task created by the mail team when mail arrives for a CAVC Appeal which is in processing with the CAVC Litigation Support
```

I think?

File: app/models/tasks/cavc_correspondence_mail_task.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+# team. May or not end up being related to CAVC response.
+#
+# Expoected Parent Task: RootTask
```
Comment (n1zyy) [inline app/models/tasks/cavc_correspondence_mail_task.rb:7]:
```suggestion
# Expected Parent Task: RootTask
```

**Review by n1zyy — APPROVED** [review]
Two trivial nits aside, this looks perfect!

**Review by leikkisa — APPROVED** [review]

**Review by n1zyy — APPROVED** [review]
Ran through the steps; all looks good!

**Pull request merged**